src/getProfile.js

```diff
@@ -0,0 +1,40 @@
+import { fetchProfiles } from './utils';
+
+const HOUR = 60 * 60
+
+export const getProfile = async (request, env, ctx) => {
+  const body = await request.json();
+  const relayUrls = body.relays.map(url => (new URL(url)).origin);
+  const pubkey = body.pubkey;
+
+  if (!relayUrls || !pubkey) {
+    return new Response('INVALID_REQUEST', { status: 422 });
+  }
+
+  const setCache = (key, data) => env.PROFILES.put(key, data, {
+    metadata: { createdAt: Date.now() },
+    // key expires after 2 days
+    expirationTtl: 2 * 24 * HOUR
+  })
+  const getCache = key => env.PROFILES.getWithMetadata(key)
```

Comment:
these helper functions could be moved outside of the `getProfile` function - I think they don't need to be within in the function body here.

this keeps the function smaller and makes it easier to read.

Reply:
But we access the PROFILES using env, that's why I didn't move them out

Reply:
true, correct.